lms/static/scripts/frontend_apps/utils/url.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Replace parameters in a URL template with values from a `params` object and
+ * returns the expanded URL.
+ *
+ *   replaceURLParams('/things/:id', {id: 'foo'}) => '/things/foo'
+ *
+ * @throws Error in case any provided param does not have a matching
+ *               placeholder in the template URL
+ */
+export function replaceURLParams<Param>(
+  urlTemplate: string,
+  params: Record<string, Param>,
+): string {
+  const paramEntries = Object.entries(params);
+  let url = urlTemplate;
+
+  paramEntries.forEach(([param, value]) => {
```

Comment:
Was there a particular reason for using `forEach` over a `for ... of` loop here?

Reply:
Mainly, not having to do the `hasOwnProperty` check on every iteration.

Reply:
oh, wait. I could still use a regular loop even with `Object.entries()` 😅. I think my brain got ahead of me.

I'll keep the loop but iterating over `Object.entries(params)` instead of `params` directly

Reply:
That isn't necessary since `Object.entries` only enumerates own properties.

Reply:
Yep. I went back to a `for ... of` loop which iterates over the result of `Object.entries`.